### DISCUSSION TITLE
Python的第三方函式庫—matplotlib(1/2)

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#203] [General]
Python的第三方函式庫，不會隨安裝Python而存在電腦中，需要藉由網路下載並安裝。如pandas、numpy與matplotlib。可以使用命令提示字元搭配python的pip install 第三方函式庫名稱，下載此第三方函式庫。因為第三方函式庫式是一個package(程式包)，其實裡面都是文字檔，只要使用匯入模組的指令—import，就可以使用此第三方函式庫。

**Comment by enoch-lightning** [comment]
而matplotlib是python用來畫圖的唯一工具，只要使用python畫圖，就一定要使用matplotlib。

**Comment by Jie2222** [comment]
最近因為其他課程需要發現matplotlib.animation 可以把數據做成動畫，滿酷的，同學有興趣也可以試試

**Reply by enoch-lightning** [reply]
了解，我會試試。謝謝妳的分享。

**Comment by enoch-lightning** [comment]
# 練習一：畫y=sin(x)
我是使用老師之前在討論區提過，使用別人電腦跑Python，但同樣使用jupyter lab介面的方式撰寫。
討論區如下
#183 
![image](https://user-images.githubusercontent.com/86505454/146715083-c2db0545-2354-49f7-b387-c6134a86a021.png)

![image](https://user-images.githubusercontent.com/86505454/146715132-c46da042-dcf3-4b9f-a647-9db3532f7032.png)
問題：
- 1.不知道粉紅色的地方怎麼會跑出來，也不知道裡面在表示什麼。
- 2.看起來中文不能當作title(標題)，正在找解決方法。
- 3.y=sin(x)的x軸想畫幾點就可以畫幾點很方便，但應該有能表示出y=sin(x)圖形的最小點數。目前還想不出來這個最小點數應該是多少。

**Reply by Jie2222** [reply]
那個粉紅色的warning會出現因該就是你標題跑不出中文的原因，matplotlib自帶的字型庫不支援中文，網路上有一些解決方法可以試試看

**Reply by enoch-lightning** [reply]
謝謝妳的回答，雖然我覺得從粉紅色的地方，其實看不出matplotlib跑不跑得出中文。

**Reply by Jie2222** [reply]
因為它有說  **missing from current font**

**Comment by enoch-lightning** [comment]
# 練習二：在自己的電腦(本機端)安裝matplotlib並作練習
影片連結如下。
[matplotlib初探](https://youtu.be/j1Dk_fqdMwg)

**Reply by enoch-lightning** [reply]
我發現我錄的影片聲音常常很不清楚，會想辦法改善。

**Reply by enoch-lightning** [reply]
@benctw 老師，想請問您聲音是怎麼錄清楚的？謝謝您。

**Reply by chia-kai-chang** [reply]
我是使用Blue Yeti麥克風，或是使用鐵三角AT2005麥克風。就會有比較好的收音。另一種方式是你可以事後使用Audacity軟體過去雜訊，並做聲音壓縮增益，也會得到比較好聽的聲音。像這個就是經過聲音壓縮增益的效果：https://youtu.be/1-pVJc4AUs8

聲音錄製是一個坑，恩~

**Reply by enoch-lightning** [reply]
謝謝老師，我如果是用我的耳機收音，可能就要用老師說的Audacity去除雜訊。這可能是最省錢的方法了。

**Reply by chia-kai-chang** [reply]
沒錯，使用手機的耳麥去收音，也可以取得不錯的聲音品質，可以再去youtube找找Audacity的使用教學，將聲音做優化，就可以得到不錯的成果了。
